corehq/apps/userreports/tasks.py

```diff
@@ -31,6 +31,8 @@ def rebuild_indicators(indicator_config_id):
         except DataError as e:
             logging.exception('problem saving document {} to table. {}'.format(doc['_id'], e))
     adapter.engine.dispose()
+    config.built = True
+    config.save()
```

Comment:
I was thinking more of being able to give more of a progress dialog to the user. You could still use this to determine if the config has been built but I would suggest setting `built = False` when changes get made to the config.